client/state/shortcodes/reducer.js

```diff
@@ -50,6 +52,32 @@ export const requesting = createReducer( {}, {
  * @return {Object}        Updated state
  */
 export const items = createReducer( {}, {
+	[ MEDIA_ITEMS_RECEIVE ]: ( state, { siteId, data } ) => {
+		if ( ! state.hasOwnProperty( siteId ) ) {
+			return state;
+		}
+
+		let media;
+		if ( Array.isArray( data.media ) ) {
+			media = data.media;
+		} else {
+			media = [ data ];
+		}
+
+		const updatedIds = media.map( ( item ) => item.ID.toString() );
+
+		return Object.assign( {}, state, {
```

Comment:
What do you think about object spread, which wasn't available in the initial implementation?

```js
return {
	...state,
	[ siteId ]: pickBy( state[ siteId ], ( shortcode ) => {
		// ...
	} )
};
```